non-exhaustive patterns: type `&FileFormat` is non-empty

Hello,

After 0.12, when I disable default features, `config` fails with the following error.

For example, for [this example](https://github.com/mehcode/config-rs/blob/master/examples/env-list/main.rs), we don't need any feature to be enabled.

```
error[E0004]: non-exhaustive patterns: type `&FileFormat` is non-empty
   --> config-rs-753d70ca56f8adb9/7cf33db/src/file/format/mod.rs:101:15
    |
34  | / pub enum FileFormat {
35  | |     /// TOML (parsed with toml)
36  | |     #[cfg(feature = "toml")]
37  | |     Toml,
...   |
57  | |     Json5,
58  | | }
    | |_- `FileFormat` defined here
...
101 |           match self {
    |                 ^^^^
    |
    = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms
    = note: the matched value is of type `&FileFormat`
    = note: references are always considered inhabited

For more information about this error, try `rustc --explain E0004`.
```

I think adding a `_ => unreachable!()` to the `match` [can do the job](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c497ec8a5eb9aa5b1e3bb79da73630ba).

--- a/src/file/format/mod.rs
+++ b/src/file/format/mod.rs
@@ -116,6 +116,16 @@ impl FileFormat {
 
             #[cfg(feature = "json5")]
             FileFormat::Json5 => json5::parse(uri, text),
+
+            #[cfg(all(
+                not(feature = "toml"),
+                not(feature = "json"),
+                not(feature = "yaml"),
+                not(feature = "ini"),
+                not(feature = "ron"),
+                not(feature = "json5"),
+            ))]
+            _ => unreachable!("No features are enabled, this library won't work without features"),
         }
     }
 }

